### ISSUE TITLE
Opening local file not possible while non synchronized SharePoint file is open in Excel

### ISSUE THREAD

**Issue by Roang-zero1** [xlwings/xlwings#1946]
#### OS
Windows 10

#### Versions of xlwings, Excel and Python (e.g. 0.11.8, Office 365, Python 3.7)

- xlwings: 0.27.10
- Office 365
- Python 3.9

#### Describe your issue (incl. Traceback!)

When I want to open a local file, but a Excel workbook from a non synchronized SharePoint is opened the local file cannot be opened.
When the non synchronized workbook is closed, xlwings works as expected.

This is problematic as the organization I am working with does not allow these workbooks to be synchronized to my computer, but I need the data from there for my xlwings integrated wokbook.

```python
Traceback (most recent call last):
  File "C:\Users\user\AppData\Local\pypoetry\Cache\virtualenvs\zeiterfassung-TMhrlkJT-py3.10\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\user\AppData\Local\pypoetry\Cache\virtualenvs\zeiterfassung-TMhrlkJT-py3.10\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\user\AppData\Local\pypoetry\Cache\virtualenvs\zeiterfassung-TMhrlkJT-py3.10\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\user\AppData\Local\pypoetry\Cache\virtualenvs\zeiterfassung-TMhrlkJT-py3.10\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\user\Documents\VersionControl\Zeiterfassung\web\views.py", line 87, in render_file
    workbook = xw.Book(workbook_path)
  File "C:\Users\user\AppData\Local\pypoetry\Cache\virtualenvs\zeiterfassung-TMhrlkJT-py3.10\lib\site-packages\xlwings\main.py", line 823, in __init__
    wb.fullname.lower() == fullname
  File "C:\Users\user\AppData\Local\pypoetry\Cache\virtualenvs\zeiterfassung-TMhrlkJT-py3.10\lib\site-packages\xlwings\main.py", line 1074, in fullname
    return self.impl.fullname
  File "C:\Users\user\AppData\Local\pypoetry\Cache\virtualenvs\zeiterfassung-TMhrlkJT-py3.10\lib\site-packages\xlwings\_xlwindows.py", line 849, in fullname
    return fullname_url_to_local_path(
  File "C:\Users\user\AppData\Local\pypoetry\Cache\virtualenvs\zeiterfassung-TMhrlkJT-py3.10\lib\site-packages\xlwings\utils.py", line 569, in fullname_url_to_local_path
    return search_local_sharepoint_path(
  File "C:\Users\user\AppData\Local\pypoetry\Cache\virtualenvs\zeiterfassung-TMhrlkJT-py3.10\lib\site-packages\xlwings\utils.py", line 708, in search_local_sharepoint_path
    raise xlwings.XlwingsError(
xlwings.XlwingsError: Couldn't find your SharePoint file locally, see: xlwings.org/error
```

#### Include a minimal code sample to reproduce the issue (and attach a sample workbook if required!)

```python
import xlwings as xw
from pathlib import Path

BASE_PATH = "C:\User\user\"
file_path = "book.xlsx"

workbook_path = Path(BASE_PATH, file_path)
print(os.fspath(workbook_path)) #C:\User\user\book.xlsx
workbook = xw.Book(workbook_path) #Error occurs here
```

**Comment by fzumstein** [comment]
Are you opening the "non-synced sharepoint file" from the SharePoint web page?

**Comment by Roang-zero1** [comment]
Yes I opened it from the web interface in my local Excel instance.

**Comment by fzumstein** [comment]
Would you be able to open it in the online version of Excel as a workaround? The only other workaround I can think of would be to have 2 local Excel instances and somehow achieve that xlwings is dealing with one and SharePoint dealing with the other one (I don't know how SharePoint picks the instance).

**Comment by Roang-zero1** [comment]
I can do that as a workaround, but I think this should still be addressed.

The reason I say this are two-fold:

1. There is no indication that the fact that a file from a SharePoint is open is interfering with the local file.
In fact I had to edit the local xlwings library to see which file was responsible.
There should be a notice that having any SharePoint file opened from a webpage will prevent xlwings from working.
Either in the console or in the https://docs.xlwings.org/en/latest/onedrive_sharepoint.html documentation.
2. The file I try to open has a completely different `name` than the one that has been opened from the SharePoint.
Wouldn't it be possible to ignore a unrelated file to the one I am opening when using the `Book` class to open a single file?
This error only happens because at https://github.com/xlwings/xlwings/blob/bc6846785d29d6511beff29e536b937ec0e5a57b/xlwings/main.py#L820 we search the open apps, to compare the `fullname`. If we compared the `name` before hand we could see that the SharePoint opened file is irrelevant. Therefore we can skip the fulnname resolution and prevent the error.

Basically my question is as follows:
I understand that the app already has the book open, but why should a non-synced SharePoint opened book that has a different `name` than the one I am trying to open prevent me from doing so.

Alternatively I would prefer a setting to deactivate the sharepoint/onenote functionality as I do not plan to use it.

**Comment by Roang-zero1** [comment]
Please correct me if I am wrong (I only skimmed the code and am not familiar with the code base), but could something like this work:

Replace
https://github.com/xlwings/xlwings/blob/bc6846785d29d6511beff29e536b937ec0e5a57b/xlwings/main.py#L820-L826

With

```python
candidates = []
for app in apps:
    for wb in app.books:
        if wb.name.lower() == os.path.split(fullname)[1].lower() and (
            wb.fullname.lower() == fullname
            or wb.name.lower() == fullname
        ):
            candidates.append((app, wb))
```

**Comment by fzumstein** [comment]
yes, that's a neat trick! thanks!
